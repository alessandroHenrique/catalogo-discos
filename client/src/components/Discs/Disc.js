import React from 'react';
import PropTypes from 'prop-types';

class Disc extends React.Component {
  state = {
    isEditing: false,
    titleInput: this.props.title,
    authorInput: this.props.author,
  };

  static propTypes = {
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    updateDisc: PropTypes.func.isRequired,
    deleteDisc: PropTypes.func.isRequired
  }

  handleTitleChange = (e) => {
    const value = e.target.value;
    this.setState(() => ({
      titleInput: value
    }));
  }

  handleAuthorChange = (e) => {
    const value = e.target.value;
    this.setState(() => ({
      authorInput: value
    }));
  }

  changeIsEditing = (e) => {
    if (e.target.innerHTML === 'Save') {
      this.props.updateDisc({
        title: this.state.titleInput,
        author: this.state.authorInput
      });
    }
    this.setState((prevState) => ({
      isEditing: !prevState.isEditing
    }));
  }

  render() {
    return (
      <li className="p-2" style={{ display: 'inline-block' }}>
        <div className="card mt-4 disc-widget text-left" style={{ width: 18 + 'rem', display: 'inline-block' }}>
          <div className="card-header disc-widget-header">
            <h5>{this.state.isEditing ? <input className="add-option__input edit-disc" type="text" onChange={this.handleTitleChange} value={this.state.titleInput} /> : <span>Disco: {this.state.titleInput}</span>}</h5>
          </div>
          <div className="card-body">
            <h6 className="card-subtitle mb-2">{this.state.isEditing ? <input className="add-option__input edit-disc" type="text" onChange={this.handleAuthorChange} value={this.state.authorInput} /> : <span>Autor: {this.state.authorInput}</span>}</h6>
            <button onClick={this.changeIsEditing} className="btn btn-primary mr-2">{this.state.isEditing ? 'Save' : 'Edit'}</button>
            <button className="btn btn-danger" onClick={() => this.props.deleteDisc()}>Delete</button>
          </div>
        </div>
      </li>
    );
  }
}

export default Disc;
