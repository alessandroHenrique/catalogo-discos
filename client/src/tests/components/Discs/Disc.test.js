import Disc from '../../../components/Discs/Disc';
import React from 'react';
import Store from '../../../store/configureStore';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

const store = Store();

describe('Disc', () => {
  it('should render a Disc', () => {

    const component = renderer.create(
      <Provider store={store}>
        <Disc
          title={'title'}
          author={'author'}
          deleteDisc={() => {}}
          updateDisc={() => {}}
        />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
