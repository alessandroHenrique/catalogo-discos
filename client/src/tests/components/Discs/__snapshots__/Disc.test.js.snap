// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disc should render a Disc 1`] = `
<li
  className="p-2"
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div
    className="card mt-4 disc-widget text-left"
    style={
      Object {
        "display": "inline-block",
        "width": "18rem",
      }
    }
  >
    <div
      className="card-header disc-widget-header"
    >
      <h5>
        <span>
          Disco: 
          title
        </span>
      </h5>
    </div>
    <div
      className="card-body"
    >
      <h6
        className="card-subtitle mb-2"
      >
        <span>
          Autor: 
          author
        </span>
      </h6>
      <button
        className="btn btn-primary mr-2"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>
</li>
`;
